lib/krikri/enrichments/split_coordinates.rb

```diff
@@ -0,0 +1,73 @@
+module Krikri::Enrichments
+  ##
+  # Splits a string given in `lat' or `long' in an edm:Place object and
+  # assigns `lat' and `long' with the split values.
+  #
+  # @example
+  #
+  #   Where val is a DPLA::MAP::Place,
+  #   and val.lat == '40.7127,74.0059',
+  #   assign val.lat = '40.7127' and val.long = '74.0059'
+  #
+  #   If long is filled in instead of lat, the values will be assigned in the
+  #   reverse order, with lat taking '74.0059' and long taking '40.7127'.
+  #
+  class SplitCoordinates
+    include Krikri::FieldEnrichment
+
+    ##
+    # Enrich a `DPLA::MAP::Place' object by splitting the string given
+    # in its `lat' or `long'.
+    #
+    # @param place [DPLA::MAP::Place]
+    # @return [DPLA::MAP::Place]
+    #
+    def enrich_value(place)
+      return place if !place.is_a? DPLA::MAP::Place
+
+      # place.lat and place.long are arrays of one value if they are assigned.
+
+      return place if place.lat.first && place.long.first  # no split required
+
+      if place.lat.first
+        latlong = coord_values(place.lat.first)
+        assign_latlong!(place, latlong.first, latlong.last)
+      elsif place.long.first
+        latlong = coord_values(place.long.first)
+        assign_latlong!(place, latlong.last, latlong.first)
+      end
+
+      place
+    end
+
+    def assign_latlong!(place, lat, long)
+      # We have to assign these one at a time because we can't do this:
+      # place.lat, place.long = [[nil], [nil]]
+      # ... which results in:
+      #     "value must be an RDF URI, Node, Literal, or a valid datatype.
+      #     See RDF::Literal. You provided nil"
+      # ... although this works fine:
+      # place.lat, place.long = [['10.0'], ['11.1']]
+      place.lat = lat
```

Comment:
I haven't verified, but I think `place.lat, place.long = ['10.0', '11.1']` and `place.lat, place.long = [nil, nil]` will both work fine.  This already seems to be assuming that you're only working with the value of `ActiveTriples::Term#first`.